internal/drivers/manager.go

```diff
@@ -355,7 +356,12 @@ func (m *Manager) Destroy(ctx context.Context, poolName, instanceID string) erro
 	if err != nil {
 		return err
 	}
-
+	stageRuntimeID := instance.Stage
+	deleteInfoMap := util.CreateEnvState().Get(stageRuntimeID)
+	if deleteInfoMap["shouldDelete"] == "false" {
```

Comment:
what happens if shouldDelete is not present in the map

Reply:
nothing, the older flow. Vm will be deleted.